cmd/skopeo/copy.go

```diff
@@ -72,16 +73,36 @@ See skopeo(1) section "IMAGE NAMES" for the expected format
 	flags.StringSliceVar(&opts.additionalTags, "additional-tag", []string{}, "additional tags (supports docker-archive)")
 	flags.BoolVarP(&opts.quiet, "quiet", "q", false, "Suppress output information when copying images")
 	flags.BoolVarP(&opts.all, "all", "a", false, "Copy all images if SOURCE-IMAGE is a list")
+	flags.StringVar(&opts.listProcessing, "list-processing", "CopySystemImage", "How to process image lists. Choose one of CopySystemImage, CopyAllImages, or CopyNoImages")
 	flags.BoolVar(&opts.removeSignatures, "remove-signatures", false, "Do not copy signatures from SOURCE-IMAGE")
 	flags.StringVar(&opts.signByFingerprint, "sign-by", "", "Sign the image using a GPG key with the specified `FINGERPRINT`")
 	flags.StringVar(&opts.digestFile, "digestfile", "", "Write the digest of the pushed image to the specified file")
 	flags.VarP(commonFlag.NewOptionalStringValue(&opts.format), "format", "f", `MANIFEST TYPE (oci, v2s1, or v2s2) to use in the destination (default is manifest type of source, with fallbacks)`)
 	flags.StringSliceVar(&opts.encryptionKeys, "encryption-key", []string{}, "*Experimental* key with the encryption protocol to use needed to encrypt the image (e.g. jwe:/path/to/key.pem)")
 	flags.IntSliceVar(&opts.encryptLayer, "encrypt-layer", []int{}, "*Experimental* the 0-indexed layer indices, with support for negative indexing (e.g. 0 is the first layer, -1 is the last layer)")
 	flags.StringSliceVar(&opts.decryptionKeys, "decryption-key", []string{}, "*Experimental* key needed to decrypt the image")
+	flags.MarkDeprecated("all", "please use --list-processing=CopyAllImages instead")
 	return cmd
 }
 
+// parseListProcessing parses the list processing selection
+// It returns the copy.ImageListSelection to use with image.Copy option
+func parseListProcessing(listProcessing string) (copy.ImageListSelection, error) {
+	switch listProcessing {
+	case "CopySystemImage":
```

Comment:
I think it would be more convenient for the Skopeo option names to be shorter, and probably not capitalized. E.g. in this case, just `system` might work.

(The Go enum values have the Copy/Image words because the values are in the full module namespace, not scoped to the enum type, like in many other languages; and it’s not that much of an issue because IDEs have pretty good autocompletion, unlike the shell, at least right now. [But compare #1500 , eventually.])

Reply:
Yes, autocompletion would be a good reason to have the options recognised by cobra. If you know how I'm happy to try it.

Reply:
At this point I think we can do without; only after we have the basic infrastructure established it would be reasonable to ask new flags to conform to established practice.

Of course if you want to figure out how this is done, that would be welcome as well :)